src/chatClient.cpp

```diff
@@ -393,6 +393,42 @@ int Client::importMessages(const char *externalDbPath)
         chatd::Chat &chat = chatroom->chat();
         karere::Id chatid = chatroom->chatid();
 
+        // Import last seen from external db
+        Id lastSeenId;
+        SqliteStmt stmtLastSeen(dbExternal, "select last_seen from chats where chatid=?");
+        stmtLastSeen << chatid;
+        stmtLastSeen.stepMustHaveData();
+        lastSeenId = stmtLastSeen.uint64Col(0);
+        chat.setLastSeenId(lastSeenId);
+
+        SqliteStmt stmtLastSeenIdx(dbExternal, "select idx from history where chatid = ? and msgid = ?");
+        stmtLastSeenIdx << chatid << lastSeenId;
+        if (stmtLastSeenIdx.step())
+        {
+            chat.setLastSeenIdx(stmtLastSeenIdx.intCol(0));
```

Comment:
This is risky. The `idx` given by the NSE to an specific `msgid` doesn't necessarily match the `idx` given by the app to the same message.